Tools/scripts/patchcheck.py

```diff
@@ -230,6 +260,9 @@ def travis(pull_request):
     c_files = [fn for fn in file_paths if fn.endswith(('.c', '.h'))]
     doc_files = [fn for fn in file_paths if fn.startswith('Doc') and
                  fn.endswith(('.rst', '.inc'))]
+    got_exception, duplicates = duplicates_found(python_files)
+    if duplicates:
+        print('Please fix the duplicate method definitions found')
```

Comment:
This message needs to link to instructions onto how to do so, more specifically including how to update the list of things that should be ignored when that is the solution (when it isn't I expect people to already figure out the problem by the 

As implemented, this could use the `ast` module from a potentially different version of python3 than the tree being compiled uses.  (unless we run patchcheck using our own interpreter?  i haven't looked)

Which means some test files are going to contain syntax that does not parse at various times.  Those should _not_ be things to list in an exception file.  Ideally we want such a tool to run under own interpreter so that it's ast is our ast.

Reply:
> Ideally we want such a tool to run under own interpreter so that it's ast is our ast.

IMHO it should be more restrictive. The interpreter used to run this tool must be built from the head of the master branch to handle the case where the grammar of the Python language is being changed and the tool is run on a file that tests this change which would raise a syntax error when compiled with an interpreter that does not support the new grammar change.

There is a similar issue when cross-compiling where a native interpreter is used during the cross-compilation to generate the sysconfig build-time data with 'generate-posix-vars' or to run setup.py. This problem actually occured when f-string where introduced in 3.6.